src/help/utils/apiSnippets.tsx

```diff
@@ -0,0 +1,312 @@
+    requestSnippetGenerator_python_requests: (request) => {
+      const { url, searchParams, method, headers, isJSON } =
+        requestExtract(request);
+
+      if (method === 'POST') {
+        return '# No automatic snippet, work in progress';
+      }
+
+      return `import requests, sys${isJSON ? ', json' : ''}
+${
+  searchParams
+    ? `
+params = ${JSON.stringify(
+        Object.fromEntries(searchParams),
+        (key, value) => {
+          if (key === 'fields') {
+            return value.split(',');
+          }
+          return value;
+        },
+        2
+      )}`
+    : ''
+}
+headers = ${JSON.stringify(Object.fromEntries(headers.entries()), null, 2)}
+base_url = "${url.origin + url.pathname}"
+
+response = requests.get(base_url, headers=headers, params=params)
+if not response.ok:
+  response.raise_for_status()
+  sys.exit()
+
+data = response.${isJSON ? 'json()' : 'text'}
+
+print(${isJSON ? 'json.dumps(data, indent=2)' : 'data'})
+`;
+    },
+  },
+};
+
+// JAVASCRIPT
+const SnippetGeneratorNodeJsPlugin: SnippetGeneratorPlugin = {
+  fn: {
+    // use `requestSnippetGenerator_` + key from config (node_native) for generator fn
+    requestSnippetGenerator_js_fetch: (request) => {
+      const { url, searchParams, method, headers, body, isJSON } =
+        requestExtract(request);
+
+      if (method === 'POST') {
+        return '// No automatic snippet, work in progress';
+      }
+
+      return `// Browser requirement: Firefox, Chrome, or Edge >= 89;
+// or, Node requirement: NodeJS >= 14.8.0, used in ES modules mode
+${
+  searchParams
+    ? `
+const searchParams = new URLSearchParams(${JSON.stringify(
+        Object.fromEntries(searchParams),
+        (key, value) => {
+          if (key === 'fields') {
+            return value.split(',');
+          }
+          return value;
+        },
+        2
+      )});`
+    : ''
+}
+const url = \`${url.origin + url.pathname}${searchParams ? `?$\{searchParams}` : ''}\`;
+
+const response = await fetch(url, {
+  headers: ${indent(JSON.stringify(Object.fromEntries(headers.entries()), null, 2))},${
+    method === 'GET'
+      ? ''
+      : `
+  method: "${method}",`
+  }${
+    body
+      ? `
+  body: new FormData(${indent(JSON.stringify(body, null, 2))}),`
+      : ''
+  }
+});
+if (!response.ok) {
+  throw new Error(\`$\{response.status}: $\{response.statusText}\`);
+}
+
+const data = await response.${isJSON ? 'json' : 'text'}();
+
+console.log(data);
+`;
+    },
+  },
+};
+
+// PERL
+const SnippetGeneratorPerlPlugin: SnippetGeneratorPlugin = {
+  fn: {
+    // use `requestSnippetGenerator_` + key from config (node_native) for generator fn
+    requestSnippetGenerator_perl: (request) => {
+      const { url, method, headers, isJSON } = requestExtract(request);
+
+      if (method === 'POST') {
+        return '# No automatic snippet, work in progress';
+      }
+
+      return `# Perl not recommended, limited support
```

Comment:
It's not really that we won't be giving support but rather to push people to use Python and to not expect support for Perl or R, first because in the team we don't really use this, and then to deter new bioinformaticians because in 2024 it might not be the best thing to learn first (especially Perl) unless they have a really good reason.
We could rephrase as "Limited support by the UniProt helpdesk in Perl" or something else that would not make it sound like we say the language itself is deprecated.

Reply:
Sounds good, maybe something like: "Note that there is limited UniProt help desk support for Perl-related questions."